power_grid_model_c/power_grid_model/include/power_grid_model/component/asym_line.hpp

```diff
@@ -0,0 +1,114 @@
+    double i_n_;
+    double base_i_;
+    ComplexTensor<asymmetric_t> _y_series_abc;
+    ComplexTensor<asymmetric_t> _y_shunt_abc;
+
+    ComplexTensor<asymmetric_t> compute_z_series_from_input(const power_grid_model::AsymLineInput& asym_line_input) {
+        ComplexTensor<asymmetric_t> z_series_abc;
+        if (is_nan(asym_line_input.r_na) && is_nan(asym_line_input.x_na)) {
+            ComplexTensor<asymmetric_t> r_matrix = ComplexTensor<asymmetric_t>(asym_line_input.r_aa, asym_line_input.r_bb, asym_line_input.r_cc, asym_line_input.r_ba, asym_line_input.r_ca, asym_line_input.r_cb);
+            ComplexTensor<asymmetric_t> x_matrix = ComplexTensor<asymmetric_t>(asym_line_input.x_aa, asym_line_input.x_bb, asym_line_input.x_cc, asym_line_input.x_ba, asym_line_input.x_ca, asym_line_input.x_cb);
+            z_series_abc = r_matrix + x_matrix * 1.0i;
+        } 
+        else {
+            ComplexTensor4 r_matrix = ComplexTensor4(asym_line_input.r_aa, asym_line_input.r_bb, asym_line_input.r_cc, asym_line_input.r_nn, asym_line_input.r_ba, asym_line_input.r_ca, asym_line_input.r_na, asym_line_input.r_cb, asym_line_input.r_nb, asym_line_input.r_nc);
+            ComplexTensor4 x_matrix = ComplexTensor4(asym_line_input.x_aa, asym_line_input.x_bb, asym_line_input.x_cc, asym_line_input.x_nn, asym_line_input.x_ba, asym_line_input.x_ca, asym_line_input.x_na, asym_line_input.x_cb, asym_line_input.x_nb, asym_line_input.x_nc);
+            ComplexTensor4 y = r_matrix + 1.0i * x_matrix;
+            z_series_abc = kron_reduction(y);
+        }
+        return z_series_abc;
+    }
+
+    ComplexTensor<asymmetric_t> compute_c_matrix_from_input(const power_grid_model::AsymLineInput& asym_line_input) {
+        ComplexTensor<asymmetric_t> c_matrix;
+        if (!is_nan(asym_line_input.c0) && !is_nan(asym_line_input.c1)) {
+            c_matrix = ComplexTensor<asymmetric_t>{(2.0 * asym_line_input.c1 + asym_line_input.c0) / 3.0, (asym_line_input.c0 - asym_line_input.c1) / 3.0 };
+        } 
+        else if (is_nan(asym_line_input.c_nn)) {
+            c_matrix = ComplexTensor<asymmetric_t>(asym_line_input.c_aa, asym_line_input.c_bb, asym_line_input.c_cc, asym_line_input.c_ba, asym_line_input.c_ca, asym_line_input.c_cb);
+        } 
+        else {
+            ComplexTensor4 c_matrix_neutral = ComplexTensor4(asym_line_input.c_aa, asym_line_input.c_bb, asym_line_input.c_cc, asym_line_input.c_nn, asym_line_input.c_ba, asym_line_input.c_ca, asym_line_input.c_na, asym_line_input.c_cb, asym_line_input.c_nb, asym_line_input.c_nc);
+            c_matrix = kron_reduction(c_matrix_neutral);
+        }
+        return c_matrix;
+    }
+
+    BranchCalcParam<symmetric_t> sym_calc_param() const override {
+        DoubleComplex y1_series_ = average_of_diagonal_of_matrix(_y_series_abc) - average_of_off_diagonal_of_matrix(_y_series_abc);
+        DoubleComplex y1_shunt_ = average_of_diagonal_of_matrix(_y_shunt_abc) - average_of_off_diagonal_of_matrix(_y_shunt_abc);
```

Comment:
local variables should not have suffix.

member data variables should have suffix.